src/main/haskell/kore/src/Kore/Variables/Free.hs

```diff
@@ -69,6 +69,9 @@ pureMergeVariables (ForallPattern f) =
     Set.insert (forallVariable f) (forallChild f)
 pureMergeVariables p = fold p  -- default rule
 
+{-| Extracts free variables in a pure pattern as a set
+of pure (not 'Unified') variables.
```

Comment:
Should we have a rule for `{-|` vs `--|` comments? I've seen both used for both short and long comments.

Reply:
I'm usually following the existing style of the file, and I prefer using `-- |` only for short comments.